src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        currentState = {
+          ...currentState, ...action.extraData,
+        };
+        result.push({ ...currentState });
+        break;
+
+      case 'removeProperties':
+        const keysToRemove = action.keysToRemove || [];
+
+        for (const key of keysToRemove) {
+          delete currentState[key];
+        }
+        result.push({ ...currentState });
+        break;
+
+      case 'clear':
+        currentState = {};
+        result.push({ ...currentState });
```

Comment:
in order not to duplicate the `result.push({ ...currentState });` you can write it only one time after `switch, case`